Frontend/src/components/AddAsset.jsx

```diff
@@ -43,30 +103,259 @@ function AddAsset({ onAdd }) {
             <span className="close" onClick={() => setShowModal(false)}>
```

Comment:
It would be better to add it as a function and call it here and anywhere that may need it.